### DISCUSSION TITLE
jump時の離床直前の制御について

### DISCUSSION THREAD

**Discussion by Naoki-Hiraoka** [YutaKojio/hrpsys-base#67] [Q&A] [answered]
jump時の離床直前は`remain_time`がほぼゼロになるので，重心・ZMPの計算が不安定になってしまうのではないかと思ったのですが，実際のところはどうなのでしょうか．

https://github.com/YutaKojio/hrpsys-base/blob/49aeca84efde2d52655e3ee6b0fb0fc1219861cb/rtc/AutoBalancer/GaitGenerator.cpp#L1037

ref軌道は離床直前の重心位置と速度が目標とほぼ一致しているので計算が不安定になることはなさそうですが，act値を用いる軌道は不安定になるかもしれないと思いました．

**Comment by YutaKojio** [comment, marked as answer]
なると思います．
ただ，１歩先を終端として歩行してた時代も振動しがちではあったけど実機でも歩けてたから，実機で不可能な動作というわけではないと思います．
ですが，jump,runもn-step対応したほうがいいのはたしかです．
